### PR TITLE
Make pilatus deadtime overridable and add tests

### LINKED ISSUE #355: Make `PilatusController` support Pilatus 3 deadtime
As a user I would like to trigger Pilatus 3 detectors at max FPS so that I can stress test them with ophyd-async. 

Currently the [Pilatus controller deadtime](https://github.com/bluesky/ophyd-async/blob/b2b8bba34eed7bd8d7ee55a594a4fcfe5f4be888/src/ophyd_async/epics/areadetector/controllers/pilatus_controller.py#L30) is given as 2.28ms, based on the Pilatus 1.4 manual. Pilatus 3s have a shorter deadtime.

## Possible Solution

There are two options:
1. Change the deadtime, only support Pilatus 3 or newer
2. Make the deadtime configurable, expect the user to pass in the model number at instantiation

## Acceptance Criteria
1. Possible to construct a pilatus detector with the correct deadtime

### PR DESCRIPTION
**Pull request by callumforrester**
Fixes #355

### CODE REVIEW

**Comment by DiamondJoseph** [comment]
Is the current Pilatus implementation compatible with a Pilatus2 such that we need to support the current value at all?

**Comment by callumforrester** [comment]
Good question, either way the Pilatus 3 manual doesn't specify a deadtime: https://media.dectris.com/user-manual-pilatus3-2020.pdf

It seems to be more detector-dependant.

**Comment by callumforrester** [comment]
Nevermind then, I can't read

**Comment by coretl** [comment]
> @coretl Shall we _just_ support the Pilatus 3?

yes, until someone else asks for an older one

File: src/ophyd_async/epics/areadetector/pilatus.py [hunk 1]
```diff
@@ -10,6 +10,12 @@
 from ophyd_async.epics.areadetector.writers.hdf_writer import HDFWriter
 from ophyd_async.epics.areadetector.writers.nd_file_hdf import NDFileHDF
 
+#: Cite: https://media.dectris.com/User_Manual-PILATUS2-V1_4.pdf
+#: The required minimum time difference between ExpPeriod and ExpTime
+#: (readout time) is 2.28 ms
+#: We provide an option to override for newer Pilatus models
+DEFAULT_READOUT_TIME_SECONDS: float = 2.28e-3
```
Comment (coretl) [inline src/ophyd_async/epics/areadetector/pilatus.py:17]:
```suggestion
class PilatusReadouttime(Enum):
    """Pilatus readout time per model in ms"""
    pilatus2 = 2.28e-3
    pilatus3 = 0.95e-3
```

**Review by coretl — COMMENTED** [review]
How about this?

**Review by DiamondJoseph — APPROVED** [review]

**Pull request merged**